tests/validation/tests/v3_api/test_node.py

```diff
@@ -1,54 +1,961 @@
+        "Label is not set on node template"
+    assert node_template_new["labels"][test_label] == new_value
+
+    # Add a node in cluster
+    cluster, node_pools = add_node_cluster(node_template_new, cluster)
+
+    nodes = client.list_node(clusterId=cluster.id).data
+    """check original label on the first node,
+    and the new label on the added node"""
+    for node in nodes:
+        if node.nodeName != existing_node.nodeName:
+            validate_label_set_on_node(client, node, test_label, new_value)
+        else:
+            validate_label_set_on_node(client, node, test_label, label_value)
+
+
+def test_node_label_node_template_delete():
+    client = cluster_detail["client"]
+    if cluster_node_template["cluster"] is None:
+        test_label = random_name()
+        label_value = random_name()
+        cluster_node_template["cluster"], node_pools, \
+            node_template, do_cloud_credential = \
+            create_cluster_node_template_label(test_label, label_value)
+        cluster_node_template["node_pools"] = node_pools[0]
+        cluster = cluster_node_template["cluster"]
+        cluster_node_template["node_template"] = node_template
+        cluster_node_template["do_cloud_credential"] = do_cloud_credential
+        cluster_node_template["test_label"] = test_label
+        cluster_node_template["label_value"] = label_value
+    else:
+        cluster = cluster_node_template["cluster"]
+        node_template = cluster_node_template["node_template"]
+        do_cloud_credential = cluster_node_template["do_cloud_credential"]
+        test_label = cluster_node_template["test_label"]
+        label_value = cluster_node_template["label_value"]
+    create_kubeconfig(cluster_node_template["cluster"])
+    nodes = client.list_node(clusterId=cluster.id).data
+    existing_node_name = []
+    # list of exiting node names
+    for node in nodes:
+        existing_node_name.append(node.nodeName)
+    # delete label in node template
+    template_label = node_template.labels.data_dict()
+    del template_label[test_label]
+
+    # update node template
+    node_template_new = client.update(node_template, labels=template_label,
+                                      cloudCredentialId=do_cloud_credential.id,
+                                      digitaloceanConfig=
+                                      {"region": "nyc3",
+                                       "size": "2gb",
+                                       "image": "ubuntu-16-04-x64"})
+    node_template_new = client.wait_success(node_template_new)
+    assert test_label not in node_template_new["labels"], \
+        "Label is available on the node template"
+
+    # Add a node in cluster with new node template
+    cluster, node_pools = add_node_cluster(node_template_new, cluster)
+
+    nodes = client.list_node(clusterId=cluster.id).data
+    """check original label on the first node,
+    and the new label on the added node """
```

Comment:
Can you get the exact node that gets added after this change and validate the check only on that node